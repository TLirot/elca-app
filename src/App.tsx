import React, { FC } from 'react';
import './App.css';
import { Card } from "./components/layout/card";
import { MenuView } from "./views/menu.view";
import { BrowserRouter, Redirect, Route, Switch } from "react-router-dom";
import { BitcoinView } from "./views/bitcoin/bitcoin.view";
import { FibonacciView } from "./views/fibonacci/fibonacci.view";
import { AppRoutesEnum } from "./routes/app.routes.enum";

const App: FC = () => {
    return (
        <Card title={"ELCA APP - TORIL CASTILLO, MIGUEL"}>
            <BrowserRouter basename={"/"}>
                <MenuView/>
                <Switch>
                    <Route path={AppRoutesEnum.fibonacci}>
                        <FibonacciView/>
                    </Route>
                    <Route path={AppRoutesEnum.bitcoin}>
                        <BitcoinView/>
                    </Route>
                </Switch>
                <Redirect to={AppRoutesEnum.bitcoin}/>
            </BrowserRouter>
        </Card>
    );
}

export default App;
