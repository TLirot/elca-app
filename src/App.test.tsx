import React from 'react';
import App from './App';
import { shallow } from 'enzyme';

describe("App", () => {
  it("renders learn react link", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  })
});
