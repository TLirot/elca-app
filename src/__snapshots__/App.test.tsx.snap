// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders learn react link 1`] = `
<Card
  title="ELCA APP - TORIL CASTILLO, MIGUEL"
>
  <BrowserRouter
    basename="/"
  >
    <MenuView />
    <Switch>
      <Route
        path="/fibonacci"
      >
        <FibonacciView />
      </Route>
      <Route
        path="/bitcoin"
      >
        <BitcoinView />
      </Route>
    </Switch>
    <Redirect
      to="/bitcoin"
    />
  </BrowserRouter>
</Card>
`;
